controllers/cartController.js

```diff
@@ -0,0 +1,19 @@
+const cartService = require("../services/cartService");
+const { catchAsync, raiseCustomError } = require("../utils/error");
+
+const addProduct = catchAsync(async (req, res) => {
```

Comment:
```suggestion
const addProduct = catchAsync(async (req, res) => {
```
`addProduct`는 상품 자체를 추가하는 의미를 가지는 것 같습니다. 장바구니 데이터를 생성하는 의미의(`addCart`) 네이밍이 필요할 것 같아요.